app/src/main/java/com/vci/vectorcamapp/intake/presentation/IntakeViewModel.kt

```diff
@@ -145,7 +145,14 @@ class IntakeViewModel @Inject constructor(
                         specimenConditionResult
                     ).any { it is Result.Error }
 
-                    if (!hasError) {
+                    if (hasError) {
+                        emitError(IntakeError.FORM_INVALID)
+                        return@launch
```

Comment:
Do we need this early return statement? Are there other code blocks that run after this if statement?